### ISSUE TITLE
Possible bug in the pretrained_word_embeddings.py example?

### ISSUE THREAD

**Issue by ksrinivs64** [keras-team/keras#7115]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [ Yes] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ Yes] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [Yes ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

The [blog post](https://blog.keras.io/using-pre-trained-word-embeddings-in-a-keras-model.html) suggests we should see 95% classification accuracy using Glove.  With the latest version of Keras and Tensor Flow I see something like:
Epoch 9/10
15998/15998 [==============================] - 221s - loss: 0.6502 - acc: 0.7682 - val_loss: 0.9075 - val_acc: 0.7112
Epoch 10/10
15998/15998 [==============================] - 219s - loss: 0.5621 - acc: 0.7997 - val_loss: 0.9073 - val_acc: 0.7167

I did try switching the backend to Theano but observed the same results.  This is a significant drop off in accuracy which could be either due to changes in TensorFlow/Theano, Keras or the dataset.  The backend seems less likely as a problem because I can replicate this drop-off with both backends.  I thought I'd report it in case its helpful to tracking a problem down in Keras.

**Comment by jsoutherland** [comment]
I am getting similar results with https://github.com/fchollet/keras/blob/master/examples/pretrained_word_embeddings.py

I wanted to point out a detail from the previous post from @ksrinivs64 . The tutorial says it should hit 95% after 2 epochs and now it is at ~70% after 10 epochs.

It would be awesome if someone that is getting 90%+ could check their output against the following:

```
Using TensorFlow backend.
Indexing word vectors.
Found 400000 word vectors.
Processing text dataset
Found 19997 texts.
Found 174105 unique tokens.
Shape of data tensor: (19997, 1000)
Shape of label tensor: (19997, 20)
Preparing embedding matrix.
Training model.
Train on 15998 samples, validate on 3999 samples
Epoch 1/10
15998/15998 [==============================] - 106s - loss: 2.4233 - acc: 0.2038 - val_loss: 1.9109 - val_acc: 0.3256
Epoch 2/10
15998/15998 [==============================] - 107s - loss: 1.5615 - acc: 0.4551 - val_loss: 1.3375 - val_acc: 0.5381
Epoch 3/10
15998/15998 [==============================] - 103s - loss: 1.2087 - acc: 0.5884 - val_loss: 1.1481 - val_acc: 0.6059
Epoch 4/10
15998/15998 [==============================] - 105s - loss: 0.9825 - acc: 0.6687 - val_loss: 1.1551 - val_acc: 0.6227
Epoch 5/10
15998/15998 [==============================] - 106s - loss: 0.8247 - acc: 0.7197 - val_loss: 1.0321 - val_acc: 0.6529
Epoch 6/10
15998/15998 [==============================] - 105s - loss: 0.7024 - acc: 0.7615 - val_loss: 1.1576 - val_acc: 0.6349
Epoch 7/10
15998/15998 [==============================] - 104s - loss: 0.5920 - acc: 0.7977 - val_loss: 0.9273 - val_acc: 0.7127
Epoch 8/10
15998/15998 [==============================] - 104s - loss: 0.5005 - acc: 0.8331 - val_loss: 0.9924 - val_acc: 0.6969
Epoch 9/10
15998/15998 [==============================] - 104s - loss: 0.4399 - acc: 0.8498 - val_loss: 0.9650 - val_acc: 0.7122
Epoch 10/10
15998/15998 [==============================] - 108s - loss: 0.3666 - acc: 0.8749 - val_loss: 1.0097 - val_acc: 0.7189
```

@fchollet Any insight? I noticed that you were the author of that blog post / tutorial.  Thanks!

**Comment by jsoutherland** [comment]
I read through the other Issues, this one shows up a few more times:

https://github.com/fchollet/keras/issues/6685 .  (closed after becoming stale)

https://github.com/fchollet/keras/issues/5185   (more info, possible improvement with how the file is opened - didn't work for me)

https://github.com/fchollet/keras/issues/3664  (recommends tensorflow - I am using tensorflow)

**Comment by jsoutherland** [comment]
I found the answer in https://github.com/fchollet/keras/issues/5826 where the change is noted as having been introduced in https://github.com/fchollet/keras/pull/5585

The newsgroups dataset originally included class/label information in the data (email/thread headers).  That text has been stripped from the dataset, which has caused a decrease in performance.

@fchollet it would help to update https://blog.keras.io/using-pre-trained-word-embeddings-in-a-keras-model.html

**Comment by aalto1** [comment]
I think that addressing the blog issue is important because it actually mislead people

**Comment by DanielDIQA** [comment]
Hi there,

I don't think that the headers where the reason for the 95% val accuracy. I was not able to achieve this accuracy even with the headers included!

I rather think that this is the reason (copied from my comment to #5185 :
The stated "95% val accuracy" are correct....if you use the binary_accuracy, which is too optimistic in this example here (sparse one-hot-encoded vectors).
Probably the author of this example used in an earlier version the 'binary_crossentropy' loss function, which returns the binary_accuracy as "accuracy" during training.
I found this stackoverflow-reply helpfull:
https://stackoverflow.com/questions/42081257/keras-binary-crossentropy-vs-categorical-crossentropy-performance

Cheers
Daniel
